app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -167,4 +202,24 @@ void onLoginClicked() {
             getPresenter().login(email, password);
         }
     }
+
+    public void onGoogleSignInClicked() {
+        if (mGoogleSignInClient == null) {
+            mToastFactory.showLong(R.string.app_unexpected_error);
+        } else {
+            Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+            showProgress();
+            startActivityForResult(signInIntent, GOOGLE_SIGN_IN);
+        }
+    }
+
+    public void handleSignInWithGoogleResult(Task<GoogleSignInAccount> task) {
+        try {
+            GoogleSignInAccount account = task.getResult(ApiException.class);
+            mToastFactory.showLong(getString(R.string.login_user_signing_with_google_account) + " " + account.getEmail());
+            onUserLoggedIn();
+        } catch (ApiException e) {
+            mToastFactory.showLong(String.format("%s %s", getString(R.string.app_unexpected_error), e.getMessage()));
```

Comment:
We can save %s %s on strings file such as app_unexpected_error_format. What do you  think?

Reply:
My bad, I think it's good to save them on strings.